src/kernel/sporadic.c

```diff
@@ -64,11 +64,15 @@ UNUSED static inline void refill_print(sched_context_t *sc)
 /* check a refill queue is ordered correctly */
 static UNUSED bool_t refill_ordered(sched_context_t *sc)
 {
+    if (isRoundRobin(sc)) {
```

Comment:
I'm confused as to why this is here? RoundRobin threads don't have refill queues and aren't passed into any of the refill queue functions that call refill_ordered as far as I know.  Would it be more accurate for this to be an assert?

Reply:
I'm probably missing something, but what would prevent `refill_unblock_check` being called on a round-robin thread?

Reply:
Nothing but `refill_unblock_check` already has this at the top:
```
    if (isRoundRobin(sc)) {
        /* nothing to do */
        return;
    }
```

Reply:
Thanks, that's what I missed. :-)

Reply:
I intentionally tried to keep any round robin logic out of sporadic.c

Reply:
There was a point at which the implementation I was using also applied to round-robin threads, and I think the constraints generally still apply to them although what I ended up with still never passed them into the sproadic.c functions.